src/d3a/d3a_core/sim_results/endpoint_buffer.py

```diff
@@ -110,14 +110,19 @@ def update_stats(self, area, simulation_status):
             "load-unit": "kWh",
             "price-energy-day": export_price_energy_day(area)
         }
+        market_type = \
```

Comment:
I would recommend to create a separate private method called "_update_cumulative_grid_trades" which will handle the update of this stat, in order to not clutter up the update_stats function. It is already done in the unmatched loads case, and the PR for the price energy day follows the same pattern.